src/main/java/Group15/Util/WorkoutAlgorithm.java

```diff
@@ -90,7 +90,7 @@ private static Exercise getValidExercise(List<BodyPart> selectedBodyParts, List<
 
             for (var bodyPart : selectedBodyParts)
             {
-                if (exercise.bodyParts.contains(bodyPart))
+                if (exercise.bodyParts.getFirst() == bodyPart)
```

Comment:
Doesnt it just look for primary bodyparts now?

Reply:
Yes, that's what you wanted